arbitrum-client/src/conversion.rs

```diff
@@ -0,0 +1,203 @@
+pub fn to_contract_proof(proof: PlonkProof) -> Result<ContractProof, ConversionError> {
+    Ok(ContractProof {
+        wire_comms: try_unwrap_commitments(&proof.wires_poly_comms)?,
+        z_comm: proof.prod_perm_poly_comm.0,
+        quotient_comms: try_unwrap_commitments(&proof.split_quot_poly_comms)?,
+        w_zeta: proof.opening_proof.0,
+        w_zeta_omega: proof.shifted_opening_proof.0,
+        wire_evals: proof
+            .poly_evals
+            .wires_evals
+            .try_into()
+            .map_err(|_| ConversionError::InvalidLength)?,
+        sigma_evals: proof
+            .poly_evals
+            .wire_sigma_evals
+            .try_into()
+            .map_err(|_| ConversionError::InvalidLength)?,
+        z_bar: proof.poly_evals.perm_next_eval,
+    })
+}
+
+/// Convert an [`ExternalTransfer`] to its corresponding smart contract type
+fn to_contract_external_transfer(
+    external_transfer: ExternalTransfer,
+) -> Result<ContractExternalTransfer, ConversionError> {
+    let account_addr: U160 =
+        external_transfer.account_addr.try_into().map_err(|_| ConversionError::InvalidUint)?;
+    let mint: U160 = external_transfer.mint.try_into().map_err(|_| ConversionError::InvalidUint)?;
+    let amount: U256 =
+        external_transfer.amount.try_into().map_err(|_| ConversionError::InvalidUint)?;
+
+    Ok(ContractExternalTransfer {
+        account_addr: Address::from(account_addr),
+        mint: Address::from(mint),
+        amount,
+        is_withdrawal: external_transfer.direction == ExternalTransferDirection::Withdrawal,
+    })
+}
+
+/// Convert a [`PublicSigningKey`] to its corresponding smart contract type
+fn to_contract_public_signing_key(
+    public_signing_key: &PublicSigningKey,
+) -> Result<ContractPublicSigningKey, ConversionError> {
+    let x = try_unwrap_scalars(&public_signing_key.x.to_scalars())?;
+    let y = try_unwrap_scalars(&public_signing_key.y.to_scalars())?;
+
+    Ok(ContractPublicSigningKey { x, y })
+}
+
+/// Convert a [`SizedValidWalletCreateStatement`] to its corresponding smart
+/// contract type
+pub fn to_contract_valid_wallet_create_statement(
+    statement: &SizedValidWalletCreateStatement,
+) -> ContractValidWalletCreateStatement {
+    let public_wallet_shares = wallet_shares_to_scalar_vec(&statement.public_wallet_shares);
+
+    ContractValidWalletCreateStatement {
+        private_shares_commitment: statement.private_shares_commitment.inner(),
+        public_wallet_shares,
+    }
+}
+
+/// Convert a [`SizedValidWalletUpdateStatement`] to its corresponding smart
+/// contract type
+pub fn to_contract_valid_wallet_update_statement(
+    statement: SizedValidWalletUpdateStatement,
+) -> Result<ContractValidWalletUpdateStatement, ConversionError> {
+    let new_public_shares = wallet_shares_to_scalar_vec(&statement.new_public_shares);
+    let external_transfer: Option<ContractExternalTransfer> =
+        if statement.external_transfer == ExternalTransfer::default() {
```

Comment:
I'd suggest adding an `is_null` or otherwise to the `ExternalTransfer` type, like what we do for orders, [balances](https://github.com/renegade-fi/renegade/blob/dfc4d1c6ff1f193da2baa9992747081cf690e3b5/circuit-types/src/balance.rs#L38-L40), and fees. It is somewhat nicer to work with, I'd imagine this functionality pops up elsewhere